### ISSUE TITLE
The "use ours" and "use theirs" buttons are not available for binary files when merging branches.

### ISSUE THREAD

**Issue by Kazbayadum** [Murmele/Gittyup#241]
Hi

I think this is a serious problem. In merge conflicts we can not select "Use theirs" for binary files, because there is not any button.

There is no way for select "theirs" version. I didn't notice this bug until recently, so it looks like I'm dragging old versions of some files from a while ago.

![imagen](https://user-images.githubusercontent.com/2800267/184690188-435fb204-ca70-4487-b1d9-e24ba5c98544.png)

Same problem here

https://github.com/gitahead/gitahead/issues/624

Thanks

**Comment by Kazbayadum** [comment]
First of all, thank you very much for your great effort in the development and continuation of this project. :1st_place_medal: 

I am sorry, but two days ago, I tried to compile from the source code in different branches of this repository and the kas repository. I'm using linux. I have to say that I couldn't find the expected "use ours / use theirs" buttons. :(

I'm not sure what happens at this point when checking or unchecking conflicting files in the "Stored Files" box. So in terms of usability this feature is probably covered, but in terms of understanding and useful I think not.

I just want to understand if I made a mistake building the project or maybe there is another easier explanation. :)

**Comment by Murmele** [comment]
I am able to reproduce it in `Gittyup v1.1.1 - Montag, 19. September 2022 - 0dca6ab`
[GittyupBinaryMergeConflict.zip](https://github.com/Murmele/Gittyup/files/9604969/GittyupBinaryMergeConflict.zip)

**Comment by Murmele** [comment]
@exactly-one-kas I think I now the reason. It was because of my proposal:

![grafik](https://user-images.githubusercontent.com/10099533/191192442-1e0dfd0b-5931-4923-aaf7-1fceb9d3f0bc.png)

We allow it only if there are multiple multiple patches, but for a binary file we don't have multiple patches and therefore the buttons are not visible.

I think we need two unittests here

**Comment by Kazbayadum** [comment]
I think it's easier to show the button allways if (isConflicted). Even if there is only one patch.

But, of course, it might be more elegant if (isConflicted && (hasMultipleHunks || isBinaryFile))

**Comment by Murmele** [comment]
I think the clean way is checking if binary and I don't think it is a lot of work

**Comment by Murmele** [comment]
Maybe it is one

**Comment by exactly-one-kas** [comment]
IIRC I think it should be zero, since all hunks get a widget
We could also change to `auto hasOneHunk = patch.count() == 1` which would even work with weird text files lol

**Comment by Murmele** [comment]
> 

You are right haha ^^ Can you try it? Above there is a test repo with a zip file and if you do a merge there you have a binary merge conflict

**Comment by Kazbayadum** [comment]
Hello :)

Sorry, but I'm not sure if you are talking to me, or talking one to each other. I think you were talking to me.

Yesterday I cloned this repo and the kas repo, but I didn't find the branch with this correction. So, I'm afraid I need a more specific indication. I'm sorry but I'm a bit newbie working with branches and repos.

:D



> > 
> 
> You are right haha ^^ Can you try it? Above there is a test repo with a zip file and if you do a merge there you have a binary merge conflict

**Comment by Murmele** [comment]
No sorry I just wrote to kas

**Comment by exactly-one-kas** [comment]
![image](https://user-images.githubusercontent.com/16035731/191950937-14da794e-83f1-4c8b-86ee-8b2bc310064b.png)
It works

**Comment by Murmele** [comment]
> ![image](https://user-images.githubusercontent.com/16035731/191950937-14da794e-83f1-4c8b-86ee-8b2bc310064b.png) It works

Sounds sou strange in german ^^

**Comment by exactly-one-kas** [comment]
Yeah the translations were implemented after all English texts were in place, but those are being generated from parts in lots of places so quite a lot of translations are pretty clunky

If I had a lot of time on hand I could rework all that so that all translation calls use full sentences, but that also means either redoing all translations (I can only provide German and barely some Dutch lol) or re-assembling the current ones from their parts in the translation files
